### DISCUSSION TITLE
How do I use cache from `useFetch()` correctly?

### DISCUSSION THREAD

**Discussion by blasdfaa** [nuxt/nuxt#21357] [Questions] [answered]
Should `useFetch()` be run twice with the same key? 
I assumed that the first call is cached and the second will just return the data. Isn't that the case? 

How then should you make a request from different components in this case?
https://codesandbox.io/p/sandbox/youthful-sanne-1ziywh?welcome=true

**Comment by warflash** [comment, marked as answer]
Perhaps `useNuxtData` is what you're looking for: https://nuxt.com/docs/api/composables/use-nuxt-data#show-stale-data-while-fetching-in-the-background

Here is an example of how someone implemented a in app caching layer on top of that composable: https://stackblitz.com/edit/github-vcbsj4?file=app.vue

**Reply by blasdfaa** [reply]
> Perhaps `useNuxtData` is what you're looking for: https://nuxt.com/docs/api/composables/use-nuxt-data#show-stale-data-while-fetching-in-the-background
> 
> Here is an example of how someone implemented a in app caching layer on top of that composable: https://stackblitz.com/edit/github-vcbsj4?file=app.vue

Using `useNuxtData` does not seem very safe. Because then I need to make sure that useFetch is done before useNuxtData. 

It would be more convenient to have a tool which could use the cache if there is one, or execute `useFetch` if there is no cache. Like your example, only without the boilerplate code

**Reply by warflash** [reply]
Well you can create a composable like `useCachedFetch` that holds that boilerplate and personalize that to your liking

**Reply by blasdfaa** [reply]
> Well you can create a composable like `useCachedFetch` that holds that boilerplate and personalize that to your liking

https://github.com/nuxt/nuxt/discussions/21357#discussioncomment-6082667

**Reply by blasdfaa** [reply]
> > Well you can create a composable like `useCachedFetch` that holds that boilerplate and personalize that to your liking


 Yeah, I think that would work for me.

**Comment by DavidDeSloovere** [comment]
If you want to cache on server, Nitro has that built-in: https://nitro.unjs.io/guide/cache
